### PR TITLE
adds events to go sdk, changelog

### PR DESCRIPTION
**Pull request by andrewpmartinez**
Example of using the new events.

```
ztx, err := ziti.NewContext(cfg)

serviceAddedChan := make(chan *rest_model.ServiceDetail, 1)

serviceAddedRemover := ztx.Events().AddServiceAddedListener(func(detail *rest_model.ServiceDetail) {
	serviceAddedChan <- detail
})
```

Tests are in the `edge` repo...as there is a circular dependency trying to run controllers for integration tests here. See draft PR here: https://github.com/openziti/edge/pull/1501

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
+# Release 0.20.48
+
+## What's New
+
+* Change log - This change log was added and is a reelection of all changes from `v0.20.0` to `v0.20.23`
```
Comment (dovholuknf) [inline CHANGELOG.md:5]: reelection  --> reflection?

File: ziti/ziti.go [hunk 2]
```diff
@@ -168,7 +175,243 @@ type ContextImpl struct {
...
+func (context *ContextImpl) AddAuthQueryListener(handler func(Context, *rest_model.AuthQueryDetail)) func() {
+	listener := func(args ...interface{}) {
+		authQuery, ok := args[0].(*rest_model.AuthQueryDetail)
+
+		if !ok {
+			pfxlog.Logger().Fatalf("could not convert args[0] to %T was %T", authQuery, args[0])
+		}
+
+		if authQuery == nil {
+			pfxlog.Logger().Fatalf("expected arg[0] was nill, unexpected")
```
Comment (dovholuknf) [inline ziti/ziti.go:334]: nill typo?

Comment (dovholuknf) [reply ziti/ziti.go:334]: nill is in here in other places too. probably worth a find/replace

**Review by plorenz — APPROVED** [review]
I don't love the kataras events library any more. I prefer something more type safe and with clearer concurrency. You do have type safe high level apis, so it doesn't matter from the consumer side.

**Pull request merged**